src/cmd_line/commandLine.ts

```diff
@@ -8,7 +8,75 @@ import * as parser from './parser';
 import * as util from '../util';
 import { VimError, ErrorCode } from '../error';
 
+class CommandLineHistory {
```

Comment:
Can you put this in it's own separate file?

Reply:
Yes.
e43686d36bbf17158c96c20533b8fe1cf771aaf1